ADIOS: change subclassing from Package to AutotoolsPackage

serbanmaerean opened the pull request:
Changed the subclassing of Adios from `Package` to `AutotoolsPackage` in order to get the autoconf files that support ppc64le.

serbanmaerean commented:
I cannot see the log for the failed test.  It appears that test #7029.2, Python 2.7 test unit, failed.  Test #7029.1 passed though, whith the test unit for Python 2.6.  How can I find out what failed?

adamjstewart commented:
I couldn't see the log either, never had that happen before. Since the unit tests passed for 2.6 and macOS, I highly doubt it was anything real. I restarted the 2.7 tests. Hopefully they'll pass this time.

serbanmaerean commented:
@adamjstewart I see all checks passed this time around.  Great.  Thanks so much for restarting the tests.

serbanmaerean commented:
@tgamblin Todd, can you pls merge this PR?  It an issue w/the invocation of some python scripts that generate testing code.  Thank you.

tgamblin commented on var/spack/repos/builtin/packages/adios/package.py:
```diff
@@ -125,6 +133,18 @@ def install(self, spec, prefix):
         sh = which('sh')
         sh('./autogen.sh')
 
+        if self.get_arch() == 'ppc64le':
```
You should be able to simplify this to `if 'target=ppc64le' in spec`

adamjstewart replied:
@tgamblin Can this, and all of the other possible values of os/target/etc be added to the documentation?

tgamblin replied:
yes, we should add a section on that.

tgamblin commented on var/spack/repos/builtin/packages/adios/package.py:
```diff
@@ -125,6 +133,18 @@ def install(self, spec, prefix):
+            # update the config.sub/guess files
+            with working_dir("config"):
+                # get new config.guess and config.sub files
+                print 'Backing up existing config.[sub|guess] files\n'
```
This shouldn't print extra output.

tgamblin commented on var/spack/repos/builtin/packages/adios/package.py:
```diff
@@ -125,6 +133,18 @@ def install(self, spec, prefix):
+                # get new config.guess and config.sub files
+                print 'Backing up existing config.[sub|guess] files\n'
+                subprocess.call("mv config.sub config.sub.orig", shell=True)
+                subprocess.call("mv config.guess config.guess.orig",
```
Just use the `move` command, an alias for `shutil.move`, which is already in the package's global scope.

tgamblin commented on var/spack/repos/builtin/packages/adios/package.py:
```diff
@@ -125,6 +133,18 @@ def install(self, spec, prefix):
+                                shell=True)
+                print 'Downloading lastest config.[sub|guess] files\n'
+                subprocess.call("wget -O config.sub 'http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD'", shell=True)
+                subprocess.call("wget -O config.guess 'http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD'", shell=True)
```
Packages shouldn't fetch things manually like this, as it isn't checksummed or trusted (this is raw `http`), and we can't re-route it to fetch from a mirror on an airgapped network.

If you must fetch something remotely, use either the [`patch()`](https://spack.readthedocs.io/en/latest/packaging_guide.html#patch) directive, which will fetch a patch and check it against a checksum, OR for other files you can use the [resource(https://spack.readthedocs.io/en/latest/packaging_guide.html#resources-expanding-extra-tarballs)] directive.

But in this case you don't have to do either.  If you change the Adios package to extend [`AutotoolsPackage`](https://spack.readthedocs.io/en/latest/packaging_guide.html#implementing-the-installation-procedure) instead of `Package`, our `AutotoolsPackage` subclass already copies in the latest `config.guess` from the system.  We've used this to build a number of packages on our `ppc64le` machines.

serbanmaerean replied:
I subclassed Adios from AutotoolsPackage instead of Package--sorry, I was not aware of this before.

adamjstewart commented:
It looks like there's a conflict now. Can you resolve this?

serbanmaerean commented:
The python fix was added by smbd else while I changed the package file.  I resolved the conflict and removed my python fix.  The only fix should now be only the change in subclassing.

adamjstewart commented:
@tgamblin Can you approve?
